text/src/swam/witx/parser/TypesParser.scala

```diff
@@ -0,0 +1,102 @@
+package swam
+package witx
+package parser
+
+import fastparse._
+import swam.witx.parser.parser.WitxWhitespace._
+import swam.witx.unresolved._
+
+import scala.collection.MapView
+
+/**
+    @author Javier Cabrera-Arteaga on 2020-03-18
+  */
+object TypesParser {
+
+  import Lexical._
+
+  private var declaredTypes = Map[String, BaseWitxType](
+    "u32" -> BasicType("u32"),
+    "u64" -> BasicType("u64"),
+    "u8" -> BasicType("u8"),
+    "u16" -> BasicType("u16"),
+    "string" -> BasicType("string")
+  )
+
+  def file[_: P]: P[Map[String, BaseWitxType]] = {
```

Comment:
Generally speaking, can you remove all useless braces? When your function is a single expression, the braces are not necessary.

Reply:
Gotcha